web/modules/custom/maybe_forms/src/Services/AccountService.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace Drupal\maybe_forms\Services;
+
+use Drupal\user\Entity\User;
+
+/**
+ * Class AccountService.
+ *
+ * The AccountService class provides methods for managing user accounts.
+ *
+ * @package Maybe
+ */
+class AccountService {
+
+  /**
+   * Creates a new user account with role.
+   *
+   * @param array $data
+   *   An associative array containing user account data:
+   *   - 'name': The username for the new user.
+   *   - 'mail': The email address for the new user.
+   *   - 'pass': The password for the new user.
+   *   - 'address': The address for the new user (optional).
+   * @param string $role
+   *   The role to assign to the new user.
+   *
+   * @return int|false
+   *   The user ID of the created user on success, or FALSE on failure.
+   *
+   * @throws \Drupal\Core\Entity\EntityStorageException
+   *   Thrown if there was an issue saving the user entity.
+   */
+  public function createAccount($data, $role) {
+    $user = User::create();
+    $user->setUsername($data['name']);
+    $user->setEmail($data['mail']);
+    $user->setPassword($data['pass']);
+    $user->enforceIsNew();
+    $user->set('field_address', $data['address']);
+    $user->addRole($role);
+    $user->save();
+    return $user->id();
+  }
+
+  /**
+   * Checks if a user with the given username exists in the system.
+   *
+   * @param string $username
+   *   The username to check for existence.
+   *
+   * @return bool
+   *   TRUE if a user with the specified username exists, FALSE otherwise.
+   */
+  public function checkUserIsExist($username) {
```

Comment:
Change this function's name, exist is not a noun.

Reply:
updated